blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticsearchRestHelper.java

```diff
@@ -0,0 +1,261 @@
+import com.rackspacecloud.blueflood.service.ElasticIOConfig;
+import com.rackspacecloud.blueflood.types.Event;
+import com.rackspacecloud.blueflood.types.IMetric;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.utils.GlobPattern;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.methods.*;
+import org.apache.http.entity.ContentType;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.nio.entity.NStringEntity;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ElasticsearchRestHelper {
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchRestHelper.class);
+    private final CloseableHttpClient closeableHttpClient;
+    private final String baseUrl;
+
+    private static final ElasticsearchRestHelper INSTANCE = new ElasticsearchRestHelper();
+
+    public static ElasticsearchRestHelper getInstance() {
+        return INSTANCE;
+    }
+
+    private ElasticsearchRestHelper(){
+        logger.info("Creating a new instance of ElasticsearchRestHelper...");
+        Configuration config = Configuration.getInstance();
+        this.baseUrl = String.format("http://%s",
+                config.getStringProperty(ElasticIOConfig.ELASTICSEARCH_HOST_FOR_REST_CLIENT));
+        this.closeableHttpClient = HttpClientBuilder.create().build();
+    }
+
+    public String fetchEvents(String tenantId, Map<String, List<String>> query) throws IOException {
+        //Example URL: localhost:9200/metric_metadata/metrics/_search";
+        String url = String.format("%s/%s/%s/_search?routing=%s", baseUrl,
+                EventElasticSearchIO.EVENT_INDEX, EventElasticSearchIO.ES_TYPE, tenantId);
+
+        HttpPost httpPost = new HttpPost(url);
+        httpPost.setHeaders(getHeaders());
+
+        String queryDslString = getDslString(query);
+        CloseableHttpResponse response = null;
+
+        try {
+            HttpEntity httpEntity = new NStringEntity(queryDslString, ContentType.APPLICATION_JSON);
+            httpPost.setEntity(httpEntity);
+
+            response = closeableHttpClient.execute(httpPost);
+            String str = EntityUtils.toString(response.getEntity());
+            EntityUtils.consume(response.getEntity());
+            return str;
+        }
+        catch(Exception e){
+            if((e instanceof HttpResponseException) && (response != null))
+                logger.error("Status code: " + response.getStatusLine().getStatusCode() + " with msg: " + e.getMessage());
+
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
It's better to "enrich" the wrapping exception with context about what operation was attempted as the first argument.

Reply:
done